base/div.jl

```diff
@@ -368,3 +368,9 @@ end
 # NOTE: C89 fmod() and x87 FPREM implicitly provide truncating float division,
 # so it is used here as the basis of float div().
 div(x::T, y::T, r::RoundingMode) where {T<:AbstractFloat} = convert(T, round((x - rem(x, y, r)) / y))
+
+# Vincent Lefèvre: "The Euclidean Division Implemented with a Floating-Point Division and a Floor"
+# https://inria.hal.science/inria-00070403
+# Theorem 1 implies that the following are exact if eps(x/y) <= 1
+div(x::Float32, y::Float32, r::RoundingMode) = Float32(round(Float64(x) / Float64(y), r))
```

Comment:
Does this result in a call to `rint` in llvm code? I'm away from computer, can't test myself.

Reply:
Yes (if you use `RoundNearest`):
```
julia> ddiv(x::Float32, y::Float32, r::RoundingMode) = Float32(round(Float64(x) / Float64(y), r))
ddiv (generic function with 1 method)

julia> @code_llvm ddiv(1f0,2f0, RoundNearest)
;  @ REPL[11]:1 within `ddiv`
define float @julia_ddiv_281(float %0, float %1) #0 {
top:
; ┌ @ float.jl:236 within `Float64`
   %2 = fpext float %0 to double
   %3 = fpext float %1 to double
; └
; ┌ @ float.jl:386 within `/`
   %4 = fdiv double %2, %3
; └
; ┌ @ float.jl:370 within `round`
   %5 = call double @llvm.rint.f64(double %4)
; └
; ┌ @ float.jl:233 within `Float32`
   %6 = fptrunc double %5 to float
; └
  ret float %6
}
```

Reply:
I have had problems with a target, which has only half and single precision floating point numbers, not implenting `llvm.rint.f64`. This is a problem with our `sin(::Float32)` and `cos(::Float32)` methods, but at least I can override those methods with calls to f32 llvm intrinsics, but with a quick search in llvm langref on the phone I couldn't find an intrinsic for this operation (I hope I missed it), which would make replacing this method much more cumbersome

Reply:
It isn't a function in most languages (and isn't in C), so it probably doesn't have an LLVM intrinsic (otherwise we would have used that).

Do we have some flag that can detect whether the target has Float64?